pkg/upgrade/1.0.0-1.1.0/v1alpha1/jiva_upgrade.go

```diff
@@ -275,52 +280,352 @@ func getPVCDeploymentsNamespace(
 	return ns, nil
 }
 
-func jivaUpgrade(pvName, openebsNamespace string) error {
+func validateSync(ctrlLabel, namespace string) error {
+	glog.Infof("Verifying replica sync")
+	quorum := false
+	ctrlList, err := podClient.WithNamespace(namespace).List(
+		metav1.ListOptions{
+			LabelSelector: ctrlLabel,
+		})
+	if err != nil {
+		return err
+	}
+	if len(ctrlList.Items) == 0 {
+		return errors.Errorf("no deployments found for %s in %s", ctrlLabel, namespace)
+	}
+	ctrlPod := ctrlList.Items[0]
+	syncedReplicas := 0
+	replicationFactor, err := strconv.Atoi(ctrlPod.Spec.Containers[0].Env[0].Value)
+	if err != nil {
+		return err
+	}
+	for syncedReplicas != replicationFactor {
+		syncedReplicas = 0
+		out, err := podClient.WithNamespace(ctrlPod.Namespace).
+			Exec(
+				ctrlPod.Name,
+				&corev1.PodExecOptions{
+					Command: []string{
+						"/bin/bash",
+						"-c",
+						"curl http://localhost:9501/v1/replicas",
+					},
+					Container: ctrlPod.Spec.Containers[0].Name,
+					Stdin:     false,
+					Stdout:    true,
+					Stderr:    true,
+				},
+			)
+		if err != nil {
+			return err
+		}
+		replicas := jivaClient.ReplicaCollection{}
+		err = json.Unmarshal([]byte(out.Stdout), &replicas)
+		if err != nil {
+			return err
+		}
+		for _, replica := range replicas.Data {
+			if replica.Mode == "RW" {
+				syncedReplicas = syncedReplicas + 1
+			}
+		}
+		if !quorum && syncedReplicas > (replicationFactor/2) {
+			glog.Infof("Synced replica quorum is reached")
+			quorum = true
+		}
+	}
+	glog.Infof("Replica syncing complete")
+	return nil
+}
+
+func jivaUpgrade(pvName, openebsNamespace string) (*utask.UpgradeTask, error) {
```

Comment:
Can the upgrade functions like this be refactored to reduce the number of lines to less than 100?